src/sentry/tasks/post_process.py

```diff
@@ -104,7 +104,13 @@ def handle_owner_assignment(project, group, event):
     from sentry.models import GroupAssignee, ProjectOwnership
 
     # Is the issue already assigned to a team or user?
-    if group.assignee_set.exists():
+    key = "assignee_exists:1:%s" % (group.id)
+    assignee_exists = cache.get(key)
+    if assignee_exists is None:
+        assignee_exists = group.assignee_set.exists()
+        # Cache for an hour if it's assigned. We don't need to move that fast.
+        cache.set(key, assignee_exists, 3600 if assignee_exists else 60)
+    if assignee_exists:
         return
 
     owner = ProjectOwnership.get_autoassign_owner(group.project_id, event.data)
```

Comment:
Is it also worth caching whether there is an owner at all for a longer period of time? Otherwise for projects that don't have owner assignment set up we're going to still be hitting this every minute.

Reply:
I've asked Mark to add caching in the method itself when he's back on Tuesday.